### PR TITLE
[IMP] Gene_EncryptionKeyManager: Add support for JSON fields

### PR DESCRIPTION
**Pull request by tedsalmon**
* Add ability to re-encrypt columns with JSON blobs in them using dot notation.
* Update docs

### CODE REVIEW

**Comment by convenient** [comment]
Hello @tedsalmon 

Well this is something I didn't anticipate. Am I correct in remembering that column names in mysql are not allowed to have a . in them? That means this should be safe?

So that I can capture a test case could you please provide me an example JSON payload? The tests for this module are very rough and ready so I'm happy to craft that myself if you can get me started

File: Console/ReencryptColumn.php [hunk 1]
```diff
@@ -116,9 +127,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
             $select = $connection->select()
                 ->from($tableName, [$identifier, "$column"])
-                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
-                ->where("$field NOT LIKE ?", "$latestKeyNumber:_:__%");
-
+                ->where("($field LIKE '%_:_:____%' OR $field LIKE '%__:_:____%')")
```
Comment (convenient) [inline Console/ReencryptColumn.php]: Would it be possibly to vary the select depending on the context?

I'd like for the specific columns to maintain their original query without the leading `%` but the json case can search a bit deeper?

**Comment by tedsalmon** [comment]
Hi @convenient !

> Well this is something I didn't anticipate. Am I correct in remembering that column names in mysql are not allowed to have a . in them? That means this should be safe?

Yeah, dots are a SQL delimiter character, thus they are not valid in schema, table or column names.

> So that I can capture a test case could you please provide me an example JSON payload? The tests for this module are very rough and ready so I'm happy to craft that myself if you can get me started

Here you go:
```
 {
     "behavior": "append",
     "import_source": "file",
     "user": "",
     "password": "0:3:RV6\/qFFHn1zU2QmpFA9cysL5fdxS17T34LeY0jwx27u\/SiTtgVmOzw==",
     "username": "0:3:r+6nYk642juvyu7+YGUl0mNmA3qbApenyl\/STX7XTh7\/SHNYzA==",
     "soap_version": "SOAP_1_1",
     "request_url": "",
 }
```

Thanks!
-Ted

File: Console/ReencryptColumn.php (hunk 1 above)
Comment (tedsalmon) [reply Console/ReencryptColumn.php]: Absolutely! I completed this in https://github.com/genecommerce/module-encryption-key-manager/pull/27/commits/ddfb0f7f2d27ac974fb2fec3bf6c3aff9c3a2ee8 

I really wanted to use `JSON_VALUE` and `JSON_INSERT`, but that was going to get ugly, and I don't want to make assumptions about what SQL engines / versions the end users are running.

**Review by convenient — APPROVED** [review]

**Comment by convenient** [comment]
Thank you for your contribution

**Pull request merged**